--- a/cfg.mk	2018-09-28 14:33:33.652414862 -0400
+++ b/cfg.mk	2018-09-28 14:33:37.916437913 -0400
@@ -17,7 +17,9 @@
 # Cause the tool(s) built by this package to be used also when running
 # commands via e.g., "make syntax-check".  Doing this a little sooner
 # would have avoided a grep infloop bug.
+ifeq ($(build_triplet), $(host_triplet))
 export PATH := $(builddir)/src$(PATH_SEPARATOR)$(PATH)
+endif
 
 # Used in maint.mk's web-manual rule
 manual_title = GNU Grep: Print lines matching a pattern
