diff -ruN binutils-2.18/bfd/config.bfd binutils-2.18.new/bfd/config.bfd
--- binutils-2.18/bfd/config.bfd	2007-08-28 10:19:33.000000000 -0700
+++ binutils-2.18.new/bfd/config.bfd	2010-02-27 15:18:45.000000000 -0800
@@ -499,6 +499,9 @@
   i[3-7]86-*-aros*)
     targ_defvec=bfd_elf32_i386_vec
     ;;
+  i[3-7]86-*-ros*)
+    targ_defvec=bfd_elf32_i386_vec
+    ;;
   i[3-7]86-*-chorus*)
     targ_defvec=bfd_elf32_i386_vec
     ;;
@@ -1349,6 +1352,9 @@
     targ_defvec=bfd_elf32_sparc_vec
     targ_selvecs=sunos_big_vec
     ;;
+  sparc-*-ros*) 
+    targ_defvec=bfd_elf32_sparc_vec
+    ;;
 #ifdef BFD64
   sparc-*-solaris2* | sparcv9-*-solaris2* | sparc64-*-solaris2*)
     targ_defvec=bfd_elf32_sparc_vec
diff -ruN binutils-2.18/config.sub binutils-2.18.new/config.sub
--- binutils-2.18/config.sub	2007-08-06 13:00:30.000000000 -0700
+++ binutils-2.18.new/config.sub	2010-02-27 15:18:45.000000000 -0800
@@ -1207,7 +1207,7 @@
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -solaris* | -sym* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
-	      | -aos* \
+	      | -aos* | -ros* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
 	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
 	      | -hiux* | -386bsd* | -knetbsd* | -mirbsd* | -netbsd* \
diff -ruN binutils-2.18/gas/configure.tgt binutils-2.18.new/gas/configure.tgt
--- binutils-2.18/gas/configure.tgt	2007-08-28 10:19:36.000000000 -0700
+++ binutils-2.18.new/gas/configure.tgt	2010-02-27 15:18:45.000000000 -0800
@@ -173,6 +173,7 @@
   i386-*-beos*)				fmt=elf ;;
   i386-*-coff)				fmt=coff ;;
   i386-*-elf)				fmt=elf ;;
+  i386-*-ros)				fmt=elf ;;
   i386-*-kaos*)				fmt=elf ;;
   i386-*-bsd*)				fmt=aout em=386bsd ;;
   i386-*-netbsd0.8)			fmt=aout em=386bsd ;;
@@ -357,6 +358,7 @@
   sparc-*-linux-*)			fmt=elf em=linux ;;
   sparc-fujitsu-none)			fmt=aout ;;
   sparc-*-elf)				fmt=elf ;;
+  sparc-*-ros)				fmt=elf ;;
   sparc-*-sysv4*)			fmt=elf ;;
   sparc-*-solaris*)			fmt=elf ;;
   sparc-*-netbsdelf*)			fmt=elf em=nbsd ;;
diff -ruN binutils-2.18/ld/configure.tgt binutils-2.18.new/ld/configure.tgt
--- binutils-2.18/ld/configure.tgt	2007-08-28 10:19:42.000000000 -0700
+++ binutils-2.18.new/ld/configure.tgt	2010-02-27 15:18:45.000000000 -0800
@@ -231,6 +231,7 @@
 			esac ;;
 i[3-7]86-*-netware)	targ_emul=i386nw ;;
 i[3-7]86-*-elf*)	targ_emul=elf_i386 ;;
+i[3-7]86-*-ros*)	targ_emul=elf_i386 ;;
 x86_64-*-elf*)		targ_emul=elf_x86_64
 			targ_extra_emuls=elf_i386
 			;;
@@ -551,6 +552,7 @@
 sparc*-*-aout)		targ_emul=sparcaout ;;
 sparc*-*-coff)		targ_emul=coff_sparc ;;
 sparc*-*-elf)		targ_emul=elf32_sparc ;;
+sparc*-*-ros)		targ_emul=elf32_sparc ;;
 sparc*-*-sysv4*)	targ_emul=elf32_sparc ;;
 sparc*-*-vxworks*)	targ_emul=elf32_sparc_vxworks ;;
 sparc64-*-freebsd* | sparcv9-*-freebsd* | sparc64-*-kfreebsd*-gnu | sparcv9-*-kfreebsd*-gnu)
diff -ruN binutils-2.18/ld/emulparams/elf32_sparc.sh binutils-2.18.new/ld/emulparams/elf32_sparc.sh
--- binutils-2.18/ld/emulparams/elf32_sparc.sh	2007-08-12 12:00:07.000000000 -0700
+++ binutils-2.18.new/ld/emulparams/elf32_sparc.sh	2010-02-27 15:18:45.000000000 -0800
@@ -3,7 +3,7 @@
 
 SCRIPT_NAME=elf
 OUTPUT_FORMAT="elf32-sparc"
-TEXT_START_ADDR=0x10000
+TEXT_START_ADDR=0x100000
 MAXPAGESIZE="CONSTANT (MAXPAGESIZE)"
 COMMONPAGESIZE="CONSTANT (COMMONPAGESIZE)"
 ALIGNMENT=8
