diff -ruN orig-src/binutils-2.21.1/bfd/config.bfd binutils-2.21.1/bfd/config.bfd
--- orig-src/binutils-2.21.1/bfd/config.bfd	2011-04-01 05:02:42.000000000 -0700
+++ binutils-2.21.1/bfd/config.bfd	2011-11-11 04:05:38.000000000 -0800
@@ -519,6 +519,12 @@
   i[3-7]86-*-aros*)
     targ_defvec=bfd_elf32_i386_vec
     ;;
+  i[3-7]86-*-ros*)
+    targ_defvec=bfd_elf32_i386_vec
+    ;;
+  sparc-*-ros*) 
+    targ_defvec=bfd_elf32_sparc_vec
+    ;;
   i[3-7]86-*-chorus*)
     targ_defvec=bfd_elf32_i386_vec
     ;;
diff -ruN orig-src/binutils-2.21.1/config.sub binutils-2.21.1/config.sub
--- orig-src/binutils-2.21.1/config.sub	2010-06-01 10:53:40.000000000 -0700
+++ binutils-2.21.1/config.sub	2011-11-11 04:05:38.000000000 -0800
@@ -1297,7 +1297,7 @@
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
-	      | -aos* | -aros* \
+	      | -aos* | -aros* | -ros* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
 	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
 	      | -hiux* | -386bsd* | -knetbsd* | -mirbsd* | -netbsd* \
diff -ruN orig-src/binutils-2.21.1/gas/configure.tgt binutils-2.21.1/gas/configure.tgt
--- orig-src/binutils-2.21.1/gas/configure.tgt	2011-04-01 05:02:42.000000000 -0700
+++ binutils-2.21.1/gas/configure.tgt	2011-11-11 04:05:38.000000000 -0800
@@ -180,6 +180,8 @@
   i386-*-beos*)				fmt=elf ;;
   i386-*-coff)				fmt=coff ;;
   i386-*-elf)				fmt=elf ;;
+  i386-*-ros)				fmt=elf ;;
+  sparc-*-ros)				fmt=elf ;;
   i386-*-kaos*)				fmt=elf ;;
   i386-*-bsd*)				fmt=aout em=386bsd ;;
   i386-*-netbsd0.8)			fmt=aout em=386bsd ;;
diff -ruN orig-src/binutils-2.21.1/ld/configure.tgt binutils-2.21.1/ld/configure.tgt
--- orig-src/binutils-2.21.1/ld/configure.tgt	2011-04-01 05:03:54.000000000 -0700
+++ binutils-2.21.1/ld/configure.tgt	2011-11-11 04:05:38.000000000 -0800
@@ -242,6 +242,8 @@
 			esac ;;
 i[3-7]86-*-netware)	targ_emul=i386nw ;;
 i[3-7]86-*-elf*)	targ_emul=elf_i386 ;;
+i[3-7]86-*-ros)	targ_emul=elf_i386 ;;
+sparc*-*-ros)		targ_emul=elf32_sparc ;;
 x86_64-*-elf*)		targ_emul=elf_x86_64
 			targ_extra_emuls="elf_i386 elf_l1om"
 			;;
