diff -ruN mpfr-3.1.2/config.sub mpfr-3.1.2-ros/config.sub
--- mpfr-3.1.2/config.sub	2013-03-13 08:38:21.000000000 -0700
+++ mpfr-3.1.2-ros/config.sub	2014-02-15 13:27:15.239596329 -0800
@@ -1348,7 +1348,7 @@
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
-	      | -aos* | -aros* \
+	      | -aos* | -aros* | -akaros* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
 	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
 	      | -hiux* | -386bsd* | -knetbsd* | -mirbsd* | -netbsd* \
